zboxcore/sdk/writemarker_mutex.go

```diff
@@ -315,9 +330,19 @@ func (wmMu *WriteMarkerMutex) LockBlobber(
 		if err != nil {
 			return
 		}
-		if shouldContinue {
-			continue
+		if !shouldContinue {
+			break
 		}
-		return
 	}
+
+	go func() {
+		for {
+			d := time.Since(requestTime)
```

Comment:
we should avoid such a recursions, when it can exit?

Reply:
It will run wmMu.LockBlobber and return immediately and so for loop is terminated. If the ctx is done then it will exit. In Line 218 it is waiting on ctx.Done.

If context is not done yet, it will refresh lock.

Reply:
this logic looks fuzzy, let's change recursion to the controlled loop, make several attempts to lock and then exit